program-analysis/echidna/advanced/on-using-cheat-codes.md

```diff
@@ -0,0 +1,53 @@
+# How and when to use cheat codes
+
+**Table of contents:**
+
+- [How and when to use cheat codes](#how-and-when-to-use-cheat-codes)
+  - [Introduction](#introduction)
+  - [Cheat codes available in Echidna](#cheat-codes-available-in-echidna)
+  - [Advise on using cheat codes](#advise-on-using-cheat-codes)
+
+## Introduction
+
+When solidity smart contract testing is performed from Solidity itself, usually requires some "help" in order to tackle some EVM/Solidity limitations. 
+Cheat code are special functions that allow to change the state of the EVM in ways that are not posible in production. These were introduced by Dapptools in
+hevm and adopted (and expanded) in other projects such as Foundry.
+
+## Cheat codes available in Echidna
+
+Since Echidna uses [hevm](https://github.com/ethereum/hevm), all the supported list of cheat code is documented here: https://hevm.dev/controlling-the-unit-testing-environment.html#cheat-codes. 
+If a new cheat code is added in the future, Echidna only needs to update the hevm version and everything should work out of the box.   
+
+As an example, this is code "simulates" the use of another sender for the external call using "prank":
```

Comment:
Suggestion:
"As an example, the `prank` cheat code is able to set the `msg.sender` value of the next external call."

Reply:
Fixed